### PR TITLE
Restructuring of documentation with some new content. Closes #1022.

### LINKED ISSUE #1022: Provide documentation for reviewing logs (CLI/Server) and altering log levels
## Specify type:
- Enhancement

### Priority:
- Medium

## Description:
Provide documentation for reviewing logs (CLI/Server) and altering log levels.

Provide location for logs both server and `qpc`.

Describe verbosity level capability for the CLI.
Describe running server with environment variables to increase log level.

___

## Acceptance Criteria:

- [ ] Verify that users can find the CLI and server logs
- [ ] Verify that a CLI user can utilize the verbosity flag based on the provided documentation
- [ ] Verify that a server user can increase the server log level based on the provided documentation

### PR DESCRIPTION
**Pull request by chambridge**
[docs.zip](https://github.com/quipucords/quipucords/files/1834391/docs.zip)
After unzipping, open `build/html/index.html` to see the generated _readthedocs_ output.

I did a decent amount of  restructuring of the content. Some was just pulling files into their own smaller focused files that are then included in multiple topics.

New content for review:

- advanced_topics.rst
- scanning.rst
- encryption.rst
- authentication.rst
- troubleshooting.rst

### CODE REVIEW

File: docs/source/authentication.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Authentication
+--------------
+This section breaks down the different authentication models used within Quipucords from interacting with the server from the command line or how the server interacts with remote systems.
```
Comment (abaiken) [inline docs/source/authentication.rst]: from the command line or how --> from the command line to how

File: docs/source/scanning.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Scanning
+--------
+Quipucords can scan several types of sources from network systems to vCenter Servers and Satellite servers.
+
+Scanning Lifecycle
+^^^^^^^^^^^^^^^^^^
+A user defines a scan, see `Working with Scans <working_with_scans.html>`_, and triggers scanning to begin. When the user triggers scanning to begin a scan job is create. When a scan job is created it is queued for processing. Due to the light-weight nature of the Quipucords server scan jobs are run serially. Scan jobs will transition from queued to started and processing will begin. A user can select to pause or cancel a scan job that is pending or running. A paused scan job can be restarted, a canceled scan job cannot. A scan job can halt due to a fatal error resulting in a *failed* status, the status message should help determine the cause of the failure. When a scan results in a *completed* status it produces results that can be viewed in an associated report.
```
Comment (abaiken) [inline docs/source/scanning.rst]: When the user triggers scanning to begin a scan job is create ---> "" created.

File: docs/source/scanning.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+Scan Job Decomposition
+^^^^^^^^^^^^^^^^^^^^^^
+A Scan Job is broken down into several tasks. If a scan job is performing an inspection several sources, two tasks are created for each source. The first task is a connection task which determines the ability to connect to the source and the number of systems that can be inspected for the defined source. The second task is an inspection task that gathers data from each of the systems. When a scan job is run all the connect tasks for all of the sources will be run prior to the execution of any of the inspection tasks.
```
Comment (abaiken) [inline docs/source/scanning.rst]: If a scan job is performing an inspection several sources --> "" an inspection of several sources

File: docs/source/troubleshooting.rst [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+===============
+The following sections provide a reference for viewing logging information from the command line and server to enable troubleshooting of various activities being performed with Quipucords.
+
+Troubleshooting the Command Line
+--------------------------------
+Logs for the command line can be found in the ``~/.local/share/qpc/`` directory with the name **qpc.log**. Logging information is captured for each command with the input arguments and request endpoint and response information to enable situational debugging. The command line supports a global flag ``-v`` to increase verbosity of the logs.
+
+Troubleshooting the Server
+--------------------------
+Logs from the server are mapped from the running container into a users specified directory. If you used the defaults found in `Selecting the Quipucords Server Configuration Options <configure.html#selecting-the-quipucords-server-configuration-options>`_ section, the directory would be ``~/quipucords/log``. You will find three log files within this directory:
```
Comment (abaiken) [inline docs/source/troubleshooting.rst]: If you used the defaults found in `link` section --> If you used the defaults found in the `link` section

File: docs/source/working_with_sources.rst [hunk 5]
```diff
@@ -119,96 +105,11 @@ To create a satellite source, use the following steps:
...
-
-    # qpc scan restart --id 1
-
-
-Logging out of the Quipucords Server
-------------------------------------
-When you log in to the server, the command retrieves a token that is used for authentication with subsequent command line interface commands. That token expires daily. In addition, the token is removed when you log out of the server.
+Editing, Listing, and Clearing Credentials
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Beyond creating credentials for use with sources you can perform several other operations. You can edit a credential for cases where passwords or passphrases need to be updated. You can list and filter credentials. For credentials that are no longer needed you can they can be removed. These interactions are described in the `Credentials section <man.html#credentials>`_ of the Command Line Usage.
```
Comment (abaiken) [inline docs/source/working_with_sources.rst]: For credentials that are no longer needed you can they can --> Credentials that are no longer needed can be removed.

Comment (chambridge) [reply docs/source/working_with_sources.rst]: done

File: docs/source/troubleshooting.rst (hunk 4 above)
Comment (chambridge) [reply docs/source/troubleshooting.rst]: done

File: docs/source/scanning.rst (hunk 3 above)
Comment (chambridge) [reply docs/source/scanning.rst]: done

File: docs/source/scanning.rst (hunk 2 above)
Comment (chambridge) [reply docs/source/scanning.rst]: done

File: docs/source/authentication.rst (hunk 1 above)
Comment (chambridge) [reply docs/source/authentication.rst]: done

**Comment by chambridge** [comment]
FYI
@quipucords/documentation

**Pull request merged**